btc-lsp/src/BtcLsp/Thread/BlockScanner.hs

```diff
@@ -3,60 +3,76 @@
 {-# LANGUAGE TypeApplications #-}
 
 module BtcLsp.Thread.BlockScanner
-  ( apply,
+  ( apply, scan
   )
 where
 
 import BtcLsp.Import
+import qualified BtcLsp.Rpc.ElectrsRpc as Rpc
+import qualified BtcLsp.Rpc.Helper as Rpc
 import qualified BtcLsp.Storage.Model.Block as Block
+import qualified Data.Map as M
+import qualified Data.Set as S
+import qualified Data.Vector as V
 import qualified Network.Bitcoin as Btc
 
 apply :: (Env m) => m ()
 apply = do
-  res <- runExceptT $ do
-    height <-
-      into @BlkHeight <$> withBtcT Btc.getBlockCount id
-    hash <-
-      from <$> withBtcT Btc.getBlockHash ($ from height)
-    prev <-
-      from <$> withBtcT Btc.getBlockHash ($ from height - 1)
-    mBlock <-
-      lift Block.getLatest
-    case mBlock of
-      Nothing ->
-        scanBlockT height hash prev
-      Just blockEnt ->
-        when (blockHash (entityVal blockEnt) /= hash) $
-          scanUntilT blockEnt height hash prev
-  whenLeft res $
-    $(logTM) ErrorS . logStr . inspect
+  _r <- runExceptT scan
   sleep $ MicroSecondsDelay 1000000
   apply
 
-scanBlockT ::
-  ( Env m
-  ) =>
-  BlkHeight ->
-  BlkHash ->
-  BlkPrevHash ->
-  ExceptT Failure m ()
-scanBlockT height hash prev = do
-  blk <- withBtcT Btc.getBlock ($ from hash)
-  --
-  -- TODO : !!!
-  --
-  lift $
-    $(logTM) ErrorS . logStr $ inspect blk
-  lift . void $
-    Block.createUpdate height hash prev
-
-scanUntilT ::
-  ( Env m
-  ) =>
-  Entity Block ->
+askElectrs :: (Env m) => Set Btc.Address -> ExceptT Failure m (Map Btc.Address MSat)
+askElectrs addrs = do
+  ab <- mapM askConfBalance (S.toList addrs)
+  pure $ M.fromList ab
+  where
+    askConfBalance addr = do
+      cb <- Rpc.confirmed <$> withElectrsT Rpc.getBalance ($ Left $ OnChainAddress addr)
+      pure (addr, cb)
+
+getBlockAddresses :: Btc.BlockVerbose -> Set Btc.Address
+getBlockAddresses blk = do
+  S.unions $
+    V.map extractAddr $
+      V.map Btc.scriptPubKey $ V.foldMap Btc.decVout $ Btc.vSubTransactions blk
+  where
+    extractAddr :: Btc.ScriptPubKey -> Set Btc.Address
+    extractAddr (Btc.StandardScriptPubKey _ _ _ _ addrs) = V.foldr S.insert S.empty addrs
+    extractAddr _ = S.empty
+
+heighToInt :: BlkHeight -> Integer
+heighToInt (BlkHeight c) = c
```

Comment:
This function is redundant, you can remove it and use `from` or `into` combinators instead